fix: prevent form submission with invalid state/business type placeholders (closes #1885)

aryanghai12 opened the pull request:
**What I Fixed**

Payment settings form was accepting dropdown placeholders ("State", "Province", "Type") as valid values and submitting them to Stripe.

**Changes Made**

**17 dropdowns fixed:**

- 13 state/province/county dropdowns (US, CA, AU, MX, AE, IE, BR) - from issue https://github.com/antiwork/gumroad/issues/1885
- 4 business type dropdowns (UAE, India, Canada, generic) - found and fixed (not in original issue)

**Technical fixes:**

- Changed dropdown defaults: || "State" → || ""
- Added value="" to placeholder options: <option disabled> → <option value="" disabled>
- Fixed critical validation bug: country in states → country.toLowerCase() in states - validation was completely broken due to case mismatch ("US" vs "us")

**Impact**

- ✅ State validation now actually works (was silently failing before)
- ✅ Business type validation now works (same bug pattern)
- ✅ Prevents invalid data being sent to Stripe

**Before Video (Individual)**
<video src="https://github.com/user-attachments/assets/16722e28-0e85-44d3-92d0-8e749589c5aa" width="400" controls></video>

**Before Video (Business)** 
<video src="https://github.com/user-attachments/assets/fd796a5d-9e70-41b9-b9da-6b80241e52e2" width="400" controls></video>

**After Video  (Individual)**
<video src="https://github.com/user-attachments/assets/81874d44-46db-4296-af0c-69be68d19e3b" width="400" controls></video>

**After Video  (Business)**
<video src="https://github.com/user-attachments/assets/8be32487-dd30-4bf5-b665-6c292f77640a" width="400" controls></video>

**After Video  (Mobile)**
<video src="https://github.com/user-attachments/assets/59e7c72c-fedf-48aa-94b4-e274f02d32c1" width="400" controls></video>

**Testing**

<img width="1067" height="176" alt="Screenshot from 2025-11-09 11-20-27" src="https://github.com/user-attachments/assets/b611db78-07cf-4e51-8ffe-fccc294597f6" />

**Tests Added**

- ✅ `does not allow saving placeholder state values` - validates individual state dropdown (16.39s)
- ✅ `does not allow saving placeholder state values for business` - validates business type dropdown (11.73s)

**AI Usage**

I used GitHub Copilot to help understand the codebase structure and identify related code patterns, but all code changes were written and reviewed by me.

Closes https://github.com/antiwork/gumroad/issues/1885

binary-koan commented:
Merged https://github.com/antiwork/gumroad/pull/1953 in the end, see my comment there

aryanghai12 commented:
@binary-koan 
**Bonus Issue Fixed (Not Yet Merged):**

 While working on #1885, I discovered and fixed an additional bug in the **Business Type** and **Business State** dropdowns that exists in the current main branch.

 **The Bug (still present in main):**  
 The code uses `value={complianceInfo.business_type || "Type"}` but the placeholder `<option disabled>Type</option>` has no explicit value attribute. When `business_type` is empty, the select tries to set value="Type" but no option matches this value, causing a value mismatch and potential form validation issues.

**My Fix (in this PR):**
- Changed `value={field || "Placeholder"}` → `value={field || ""}`  
- Added explicit empty value: `<option value="" disabled>Placeholder</option>`

This ensures the select element's value properly matches the disabled placeholder option when empty, preventing undefined behavior and improving form validation reliability.

**Scope:** Applied to business_type dropdown (4 instances across UAE/India/Canada/US) and business_state dropdown (3 instances across US/Canada/Australia).

**Note:** This bonus fix is included in my PR but was not part of the main issue #1885 and is not present in the current main branch.
